src/main/java/uk/gov/digital/ho/pttg/application/namematching/candidates/AliasSurnameCombinations.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.digital.ho.pttg.application.namematching.candidates;
+
+import uk.gov.digital.ho.pttg.application.namematching.CandidateName;
+import uk.gov.digital.ho.pttg.application.namematching.InputNames;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static uk.gov.digital.ho.pttg.application.namematching.candidates.AliasSurnameCombinationsFunctions.*;
+
+public class AliasSurnameCombinations implements NameMatchingCandidateGenerator {
+
+    @Override
+    public List<CandidateName> generateCandidates(InputNames inputNames) {
+        List<CandidateName> candidateNames = new ArrayList<>();
+
+        if (inputNames.aliasSurnames().isEmpty()) {
+            return candidateNames;
+        }
+
+        for (String firstName : inputNames.firstNames()) {
+            candidateNames.add(new CandidateName(firstName, inputNames.fullLastName()));
+        }
```

Comment:
I thought this was done by a different candidate generator?

Reply:
It probably is but the ordering for alias matching is very specific and precedes all the multi-name, combinations and special character stuff.  This keeps the alias rules self-contained and deduplication will make sure we don't repeat.

Reply:
However it shouldn't be doing the full last name but rather each individual word in turn - an oversight that I'll correct.